cfme/automate/dialogs/dialog_element.py

```diff
@@ -159,7 +160,8 @@ def reorder_elements(self, add_element, second_element, element_data):
             second_element: The second element to be added to the dialog.
             element_data: Already existing first element's data.
         """
-        view = navigate_to(self, 'Edit')
+        navigate_to(self.dialog, 'Edit')
+        view = self.create_view(EditElementView)
```

Comment:
We shouldn't have to create the view after navigation, it should be returned by navigate_to.  This tells me there is a strange navigation pattern going on in the framework.

This looks like a method that should exist on the `Dialog` entity, not here on the `Element` itself.  Is this editing the dialog, or editing the element?  Why are we expecting an `EditElementView` when we navigated to `Dialog` 'Edit' destination.

Reply:
@mshriver method better to called widget which is responsible for `element` edit is part of `element` view. but upto now I understand that to edit element
1. need to edit dialog
2. from dialog edit page only we need to edit element. 

so element edit facility should be part of `DialogEditView`. I just avoid this to add normal fix :D. I thing FA owner should take care of major changes related to view. but I am on it. let me try.

Reply:
Gotcha, yeah I would rather actually fix the view structure problem, either you or the FA owner (ideally working together! :) ) taking on that task of better structuring these views. 

Sounds like EditElementView just needs to be nested in DialogEdit view, or that a conditional view needs to be used.